main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -118,10 +118,13 @@ object OccurrenceAnalyzer {
     case Expression.Var(sym, tpe, loc) => (OccurrenceAst.Expression.Var(sym, tpe, loc), Map(sym -> Once))
 
     case Expression.Closure(sym, freeVars, tpe, loc) =>
-      val fv = freeVars.map {
-        case LiftedAst.FreeVar(sym, tpe) => OccurrenceAst.FreeVar(sym, tpe)
-      }
-      (OccurrenceAst.Expression.Closure(sym, fv, tpe, loc), Map.empty)
+      val (fv, o) = freeVars.foldRight[(List[OccurrenceAst.FreeVar], Map[Symbol.VarSym, Occur])]((List.empty, Map.empty))((fv, acc) =>
```

Comment:
If you write `case ...` then you can pattern match directly on `acc` instead of writing `._1` which no ones knows what is :)